plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/MultiParentCompileHook.java

```diff
@@ -113,12 +114,27 @@ private void copy(Path eslintrc, File pluginFolder) {
         return mconfig;
     }
 
-    private void compile(MavenRunner.Config mavenConfig, File path) throws PomExecutionException, IOException {
+    private void compile(MavenRunner.Config mavenConfig, File path, String parentFolder, String pluginName) throws PomExecutionException, IOException {
+        if (StringUtils.isBlank(parentFolder)) {
+            runner.run(mavenConfig, path, setupCompileResources(path), "clean", "process-test-classes", "-Dmaven.javadoc.skip");
+        } else {
```

Comment:
Only do this if the checkout is in fact a snapshot. https://github.com/jenkinsci/bom/issues/657#issuecomment-970535269 Otherwise in the normal case (any plugin with MRP and not using a local checkout override) you will be `install`ing artifacts into the local repo on top of whatever would otherwise be downloaded from Artifactory, which is risky. https://issues.apache.org/jira/browse/MINSTALL-126 / https://issues.apache.org/jira/browse/MNG-3655